Core/clim-core/presentation-translators.lisp

```diff
@@ -78,25 +78,19 @@ and used to ensure that presentation-translators-caches are up to date.")
                             :initform (make-hash-table :test #'eq)
                             :documentation "Holds transators with a simple
   from-type (i.e. doesn't contain \"or\" or \"and\").")
-   (translator-cache-generation :accessor translator-cache-generation
-                                :initform 0)
-   (presentation-translators-cache
-    :writer (setf presentation-translators-cache)
-    :initform (make-hash-table :test #'equal))))
+   (cache-gen :initform 0)
+   (cache :initform (make-hash-table :test #'equal))))
 
 (defun invalidate-translator-caches ()
   (incf *current-translator-cache-generation*))
 
-(defgeneric presentation-translators-cache (table))
-
-(defmethod presentation-translators-cache ((table translator-table))
-  (with-slots ((cache presentation-translators-cache)
-               (generation translator-cache-generation))
-      table
-    (unless (or (= generation *current-translator-cache-generation*)
+(defun presentation-translators-cache (table)
+  (check-type table translator-table)
+  (with-slots (cache cache-gen) table
+    (unless (or (= cache-gen *current-translator-cache-generation*)
                 (zerop (hash-table-size cache)))
       (clrhash cache))
-    (setf generation *current-translator-cache-generation*)
+    (setf cache-gen *current-translator-cache-generation*)
```

Comment:
Maybe use the accessor here instead of deleting it? `with-slots` generates a `slot-value` call for each use of `cache`.

Reply:
changed